icechunk/src/asset_manager.rs

```diff
@@ -823,7 +823,7 @@ mod test {
     async fn test_dont_fetch_asset_twice() -> Result<(), Box<dyn std::error::Error>> {
         // Test that two concurrent requests for the same manifest doesn't generate two
         // object_store requests, one of them must wait
-        let storage: Arc<dyn Storage + Send + Sync> = new_in_memory_storage()?;
+        let storage: Arc<dyn Storage + Send + Sync> = new_in_memory_storage().await?;
```

Comment:
there is probably some documentation we need to change like this too?

Reply:
Only the rust side is async for this, i wrapped it on the python side for now